pkg/providers/v1/aws.go

```diff
@@ -3964,19 +3964,33 @@ func (c *Cloud) buildNLBHealthCheckConfiguration(svc *v1.Service) (healthCheckCo
 			UnhealthyThreshold: 2,
 		}
 	}
-	if parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckProtocol, &hc.Protocol) {
+
+	var pathModified bool
+	protocolModified := parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckProtocol, &hc.Protocol)
+	if protocolModified {
 		hc.Protocol = strings.ToUpper(hc.Protocol)
 	}
+
 	switch hc.Protocol {
 	case elbv2.ProtocolEnumHttp, elbv2.ProtocolEnumHttps:
-		parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckPath, &hc.Path)
+		pathModified = parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckPath, &hc.Path)
 	case elbv2.ProtocolEnumTcp:
 		hc.Path = ""
 	default:
 		return healthCheckConfig{}, fmt.Errorf("Unsupported health check protocol %v", hc.Protocol)
 	}
 
-	parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckPort, &hc.Port)
+	portModified := parseStringAnnotation(svc.Annotations, ServiceAnnotationLoadBalancerHealthCheckPort, &hc.Port)
+
+	// For a non-local service, we override the health check to use the kube-proxy port when no other overrides are provided.
+	// The kube-proxy port should be open on all nodes and allows the health check to check the nodes ability to proxy traffic.
```

Comment:
> The kube-proxy port should be open on all nodes and allows the health check to check the nodes ability to proxy traffic.

My guess is that the assumption is not correct and the 10256 port is blocked by kOps.
You can get an idea of what security group rules are created by default:
https://github.com/kubernetes/kops/blob/master/tests/integration/update_cluster/aws-lb-controller/kubernetes.tf

Reply:
Reviewing the TF you've linked, I can't see where ingress is allowed for any nodeport to the workers, unless I've misread, it looks like users have to manually allow the ports to the worker nodes to allow Service Of Type Load Balancer already? 

My comment here was more, every node will open this port, because every node should be running kube-proxy or an equivalent substitute. Whether the security group rules are configured or not wasn't intended to be captured

Reply:
Nodes open ports to each other, but not to any other IP. I think currently AWS CCM add the necessary rules for node ports, so port 10256 should be added there.
We can also test by opening node ports during testing, by adding `--set cluster.spec.nodePortAccess=0.0.0.0/0` here:
https://github.com/kubernetes/cloud-provider-aws/blob/d2b28ccf0c10e75b9b90a7e5c74b2a2d98966fb7/hack/e2e/run.sh#L131

Reply:
Ack that's helpful, I'll add that hack for now and investigate updating to set up the rules correctly via CCM

Reply:
Adding this nodePortAccess seems to have not worked, latest [CI result ](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-aws/622/pull-cloud-provider-aws-e2e/1688512872096731136) suggests that is creating a duplicate rule?

Reply:
Hmm, seems that the security group for the ELB is creating an allow all rule anyway, do you agree?
https://github.com/kubernetes/cloud-provider-aws/blob/d0551093673e8c355db17249b8f069767c014748/pkg/providers/v1/aws.go#L4637-L4641
I spun up a cluster and can see that all traffic is allowed between the load balancer security group and the worker security group, the health checks are working, so perhaps theres some other issue at play with the E2E?

Reply:
> Adding this nodePortAccess seems to have not worked, latest [CI result ](https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/cloud-provider-aws/622/pull-cloud-provider-aws-e2e/1688512872096731136) suggests that is creating a duplicate rule?

You are right, not sure how we can work around that.

> Hmm, seems that the security group for the ELB is creating an allow all rule anyway, do you agree?

I see you are looking at the Classic LB related function. Look for `updateInstanceSecurityGroupsForNLB()`.

> I spun up a cluster... 

How did you do that?

Reply:
I looked at the NLB, and tested it live, the NLB version does the same thing security group wise.

> How did you do that?

Not using KOPS, but I had an AWS cluster that I could test this all on by other means.
I'll see if I can use the script for the run e2e job to spin up a cluster using KOPS